File: stacks/typescript/src/core/greeting.test.ts
import { describe, expect, it } from "vitest";
import { greet } from "./greeting.js";

describe("greet", () => {
  it("greets a name", () => {
    expect(greet("Ada")).toBe("Hello, Ada.");
  });

  it("rejects an empty name", () => {
    expect(() => greet("")).toThrow("name is required");
  });
});

File: stacks/go/src/core/greeting_test.go
package core

import "testing"

func TestGreetsAName(t *testing.T) {
	got, err := Greet("Ada")
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	if got != "Hello, Ada." {
		t.Errorf("got %q, want %q", got, "Hello, Ada.")
	}
}

func TestRejectsEmptyName(t *testing.T) {
	if _, err := Greet(""); err == nil {
		t.Error("expected error for empty name")
	}
}
